app.js

```diff
@@ -66,6 +67,17 @@ if (app.get('env') === 'production') {
 app.use(session(session_opts));
 
 //==============================================================================
+// CSRF MIDDLEWARE
+//==============================================================================
+
+app.use(cookieParser());
+
+app.use((err, req, res, next) => {
```

Comment:
passing 4 params makes it an error handling middleware, you should remove the err param

http://expressjs.com/en/guide/using-middleware.html (Error-handling middleware) section